src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -98,23 +99,26 @@ private IEnumerable<CompletionItem> GetDeclarationCompletions(BicepCompletionCon
                 }
             }
 
-            if (context.Kind.HasFlag(BicepCompletionContextKind.NestedResourceDeclarationStart))
+            if (context.Kind.HasFlag(BicepCompletionContextKind.NestedResourceDeclarationStart) && context.EnclosingDeclaration is ResourceDeclarationSyntax resourceDeclarationSyntax)
             {
                 yield return CreateKeywordCompletion(LanguageConstants.ResourceKeyword, "Resource keyword", context.ReplacementRange);
 
-                // leaving out the API version on this, because we expect its more common to inherit from the containing resource.
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Nested resource with defaults", @"resource ${1:Identifier} '${2:Type}' = {
-  name: $3
-  properties: {
-    $0
-  }
-}", context.ReplacementRange);
-
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Nested resource without defaults", @"resource ${1:Identifier} '${2:Type}' = {
-  name: $3
-  $0
-}
-", context.ReplacementRange);
+                TypeSymbol typeSymbol = model.GetTypeInfo(resourceDeclarationSyntax);
```

Comment:
Rather than using the 'name' of the type, it would be better to copy the following logic to access the `ResourceTypeReference`: 
https://github.com/Azure/bicep/blob/5df44dd5c935ebd5f73af560831bd8e122e947f8/src/Bicep.LangServer/Completions/BicepCompletionProvider.cs#L187-L188

Reply:
Updated here:
https://github.com/Azure/bicep/commit/0c0c664878dbd906644c181a547c42f854f4d5f3